backend/src/main.rs

```diff
@@ -0,0 +1,57 @@
+#![allow(clippy::multiple_crate_versions)]
+
+use axum::{routing::get, Router};
+use backend::{prelude::*, root, services::api_router};
+use cli::Parser;
+use error_stack::Report;
+use std::path::PathBuf;
+use tower::ServiceBuilder;
+use tower_http::cors::CorsLayer;
+use tracing::Level;
+
+#[tokio::main]
+#[allow(clippy::unwrap_used, clippy::expect_used)]
+async fn main() -> Result<(), InitServerError> {
+    let config: cli::Config = cli::Args::parse().try_into().map_err(|e| {
+        Report::new(InitServerError).attach_printable(format!("couldn't get config file: {e}"))
```

Comment:
You lost the original error from stack (you convert it to string). It is better to use `.change_context(..)` on `Result<,>`. In this way the original error will be inside the stack